services/rfq/relayer/pricer/fee_pricer.go

```diff
@@ -147,11 +171,92 @@ func (f *feePricer) GetDestinationFee(parentCtx context.Context, _, destination
 		fee = new(big.Int).Add(fee, l1Fee)
 		span.SetAttributes(attribute.String("l1_fee", l1Fee.String()))
 	}
+
 	span.SetAttributes(attribute.String("destination_fee", fee.String()))
 	return fee, nil
 }
 
-func (f *feePricer) GetTotalFee(parentCtx context.Context, origin, destination uint32, denomToken string, isQuote bool) (_ *big.Int, err error) {
+// addZapFees incorporates the cost of the call and the call value into the fee.
+// Note that to be conservative, we always use the QuoteFixedFeeMultiplier over the RelayFixedFeeMultiplier.
+//
+//nolint:gosec
+func (f *feePricer) addZapFees(ctx context.Context, destination uint32, denomToken string, quoteRequest *reldb.QuoteRequest, fee *big.Int) (*big.Int, error) {
+	span := trace.SpanFromContext(ctx)
+
+	if len(quoteRequest.Transaction.ZapData) > 0 {
+		gasEstimate, err := f.getZapGasEstimate(ctx, destination, quoteRequest)
+		if err != nil {
+			return nil, err
+		}
+		callFee, err := f.getFee(ctx, destination, destination, int(gasEstimate), denomToken, true)
+		if err != nil {
+			return nil, err
+		}
+		fee = new(big.Int).Add(fee, callFee)
+		span.SetAttributes(attribute.String("call_fee", callFee.String()))
+	}
+
+	if quoteRequest.Transaction.ZapNative != nil && quoteRequest.Transaction.ZapNative.Sign() > 0 {
+		callValueFloat := new(big.Float).SetInt(quoteRequest.Transaction.ZapNative)
+		valueDenom, err := f.getDenomFee(ctx, destination, destination, denomToken, callValueFloat)
+		if err != nil {
+			return nil, err
+		}
+		valueScaled, err := f.getFeeWithMultiplier(ctx, destination, true, valueDenom)
+		if err != nil {
+			return nil, err
+		}
+		fee = new(big.Int).Add(fee, valueScaled)
+		span.SetAttributes(attribute.String("value_scaled", valueScaled.String()))
+	}
+
+	return fee, nil
+}
+
+// cache so that we don't have to parse the ABI every time.
+var fastBridgeV2ABI *abi.ABI
+
+const methodName = "relay0"
+
+func (f *feePricer) getZapGasEstimate(ctx context.Context, destination uint32, quoteRequest *reldb.QuoteRequest) (gasEstimate uint64, err error) {
+	client, err := f.clientFetcher.GetClient(ctx, big.NewInt(int64(destination)))
+	if err != nil {
+		return 0, fmt.Errorf("could not get client: %w", err)
+	}
+
+	if fastBridgeV2ABI == nil {
+		parsedABI, err := abi.JSON(strings.NewReader(fastbridgev2.IFastBridgeV2MetaData.ABI))
+		if err != nil {
+			return 0, fmt.Errorf("could not parse ABI: %w", err)
+		}
+		fastBridgeV2ABI = &parsedABI
+	}
+
+	encodedData, err := fastBridgeV2ABI.Pack(methodName, quoteRequest.RawRequest, f.relayerAddress)
```

Comment:
Seeing an issue here. I don't think `quoteRequest.RawRequest` is getting set at all when the active quote is generated:

https://github.com/synapsecns/sanguine/blob/f93b3b375df843f372f7682aff9edb3de9da190d/services/rfq/relayer/quoter/quoter.go#L374-L394

Therefore we don't correctly estimate the gas here. Constructing the full request is somewhat tricky here (e.g. we need to set the destination amount in advance), happy to discuss the details on a call.

Reply:
So we actually do supply the `RawRequest` when doing the `EstimateGas()` call here:

https://github.com/synapsecns/sanguine/blob/feat/arb-call-fee-pricer/services/rfq/relayer/pricer/fee_pricer.go#L235

The `QuoteInput.QuoteRequest` is really only used to communicate the extra data around the quoter functions. I considered introducing a new struct for this, maybe simpler to just include these two fields individually within the `QuoteInput` struct?

Reply:
We supply it, but we never actually set the value for it, so it's always an empty bytes array. Let's figure out the best way to generate it, as the current contracts don't expose the encoding function.